yt/frontends/openPMD/data_structures.py

```diff
@@ -30,6 +30,19 @@
 import os
 
 
+class openPMDBasePath:
+   def  _setBasePath(self, handle):
+       dataPath = handle.attrs["basePath"][:-3]
```

Comment:
`-3` assumes `basePath` is `something/%T/` in a very obfuscated way.
it's a bit more complicated to implement it in the correct, general way (e.g., what about `/data/steps_%T_something/moreConst/` as `basePath` -> needs regex mathing at this point).

Luckily, in `1.X` of the standard this path [is fixed to `/data/%T/`](https://github.com/openPMD/openPMD-standard/blob/1.0.0/STANDARD.md#hierarchy-of-the-data-file).

Therefore please implement this as follows to make it transparent
```python
class openPMDBasePathException(Exception) :
    pass

class openPMDBasePath :
    def  _setBasePath(self, handle) :
    """
    Set the base path for the first iteration found in the file.
    TODO implement into distinct methods:
        - __init__(self, handle)
        - getIterations(self)
        - getBasePath(self, iteration)
    """
    # basePath is fixed in openPMD 1.X to `/data/%T/`
    dataPath = u"/data"

    # if the file messed up the base path we avoid throwing a cluttered
    # exception below while looking for iterations:
    if handle.attrs["basePath"].decode("utf-8") != u"/data/%T/" :
        raise openPMDBasePathException("openPMD: basePath is non-standard!")

    list_iterations = []
    for i in list(handle[dataPath].keys()) :
        list_iterations.append(i)

    # We found no iterations in basePath
    if len(list_iterations) == 0 :
        raise openPMDBasePathException("openPMD: no iterations found in basePath!")

    # just handle the first iteration found
    self.basePath = "{}/{}/".format(dataPath, list_iterations[0])
```

I also added python3 compatibility for the iteration finding (`keys()` is not iterable any more).